### PR TITLE
Port preferences to libadwaita

### PR DESCRIPTION
**Pull request by oscfdezdz**
See commit messages.

**Edit:** The extension works as expected in GNOME Shell 42.rc, but the preferences window didn't look good.

image

### CODE REVIEW

File: display-brightness-ddcutil@themightydeity.github.com/metadata.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "gettext-domain": "display-brightness-ddcutil",
     "version": 25,
     "shell-version": [
-        "40",
-        "41"
```
Comment (daitj) [inline display-brightness-ddcutil@themightydeity.github.com/metadata.json:11]: You cannot remove the old one, as it works in 40 and 41 as well. You can just add 42

Comment (daitj) [reply display-brightness-ddcutil@themightydeity.github.com/metadata.json:11]: Is there any other reason than to make it look nice?

**Comment by daitj** [comment]
What is the benefit of porting to libadwaita. Is there any other reason than to make it look nice, I haven't looked into it?  

I will do some test before merging it and the publish.

File: display-brightness-ddcutil@themightydeity.github.com/metadata.json (hunk 1 above)
Comment (oscfdezdz) [reply display-brightness-ddcutil@themightydeity.github.com/metadata.json:11]: Libadwaita was originally planned for version 41 but was delayed to 42, so I don't expect these changes to work in 40 and 41.

We can have two `prefs.ui` and check for Shell's version as extensions supporting versions 3.3X do for GTK3 and GTK4 ui files. Another way would be to have two branches and publish updates with different version numbers.
In any case, we would have to adjust both `prefs.ui` to add any new features.

**Comment by oscfdezdz** [comment]
> What is the benefit of porting to libadwaita. Is there any other reason than to make it look nice, I haven't looked into it?

I'm not sure of all the benefits it brings, at first I thought it would be [required](https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2012) to work in version 42 since the GNOME Extensions application is now using libadwaita, and the extensions preferences are now supposed to be preference pages of that window.

For now, in 42.beta, it works without porting, although libadwaita widgets simplifies a lot the UI file, as you can tell from the diff.

This is how it looks with latest released version:

gnome-display-brightness-ddcutil

**Comment by daitj** [comment]
Lets keep this PR open for now, once 42 gets a proper release in other distros as well we can merge this and publish for 42 only. We do not need to have two versions. 
I think this should be fine because our current state is quite stable for older version of gnome.

File: display-brightness-ddcutil@themightydeity.github.com/prefs.js [hunk 2]
```diff
@@ -133,15 +129,15 @@ const PrefsWidget = GObject.registerClass({
...
-}
-
-function buildPrefsWidget() {
     const preferences = new PrefsWidget();
     preferences.connect('notify::root', () => {
         const window = preferences.get_root();
-        const headerbar = new Headerbar();
-        window.set_titlebar(headerbar);
+        window.set_default_size(500, 700);
+        window.search_enabled = true;
```
Comment (oscfdezdz) [inline display-brightness-ddcutil@themightydeity.github.com/prefs.js:136]: After rebasing this PR with latest changes, these properties are no longer respected and translations are not applying.

Comment (oscfdezdz) [reply display-brightness-ddcutil@themightydeity.github.com/prefs.js]: Fixed using the new `fillPreferencesWindow` function.

**Comment by daitj** [comment]
Are you dont with the fixes? Since 4.2 is officially released, I can start testing this.

**Comment by oscfdezdz** [comment]
> Are you dont with the fixes? Since 4.2 is officially released, I can start testing this.

Yes, it is working!

**Pull request merged**